include/seqan3/alphabet/composition/cartesian_composition.hpp

```diff
@@ -76,14 +77,75 @@ namespace seqan3
  * \sa qualified
  * \sa masked
  */
-
 template <typename derived_type,
           typename first_alphabet_type,
           typename ...alphabet_types>
-      requires semi_alphabet_concept<first_alphabet_type> && (semi_alphabet_concept<alphabet_types> && ...)
-struct cartesian_composition :
+    requires semi_alphabet_concept<first_alphabet_type> && (semi_alphabet_concept<alphabet_types> && ...)
+class cartesian_composition :
     public pod_tuple<first_alphabet_type, alphabet_types...>
 {
+private:
+    //!\brief The base type of this class.
+    using base_t = pod_tuple<first_alphabet_type, alphabet_types...>;
+
+    //!\brief A meta::list The types of each composite in the composition
+    using composites = meta::list<first_alphabet_type, alphabet_types...>;
+
+    //!\brief Is set to `true` if the type is contained in the type list.
+    template <typename type>
+    static constexpr bool is_composite =
+        meta::in<composites, type>::value;
+
+    //!\brief Is set to `true` if the type is uniquely contained in the type list.
+    template <typename type>
+    static constexpr bool is_unique_composite =
+        is_composite<type> &&
+        (meta::find_index<composites, type>::value == meta::reverse_find_index<composites, type>::value);
+
+    /*!\brief 'Callable' helper class that is invokable by meta::invoke.
+      * Returns an std::true_type if the `type` is assignable from `T`.
+      */
+    template <typename T>
+    struct assignable_from
+    {
+        //!\brief The returned type when invoked.
+        template <typename type>
+        using invoke = std::conditional<assignable_concept<type, T>,
+                                        std::true_type, std::false_type>;
+    };
+
+    /*!\brief 'Callable' helper class that is invokable by meta::invoke.
+      * Returns an std::true_type if the `type` is weakly equality comparable to `T`.
+      */
+    template <typename T>
+    struct weakly_equality_comparable_with
+    {
+        //!\brief The returned type when invoked.
+        template <typename type>
+        using invoke = std::conditional<weakly_equality_comparable_with_concept<type, T>,
+                                        std::true_type, std::false_type>;
+    };
+
+    /*!\brief 'Callable' helper class that is invokable by meta::invoke.
+      * Returns an std::true_type if the `type` is weakly equality comparable to `T`.
```

Comment:
well it's merely using the `weakly_equality_comparable_with` concept so why should the documentation be different?

Reply:
So it's just an alias and they do the same thing? Then you might as well use `\copydoc`

But it's using `detail::weakly_ordered_with_concept` instead of `weakly_equality_comparable_with_concept`

Reply:
It is not an alias but a wrapper. It is a helper struct (named `callable` by [eric niebler](https://ericniebler.github.io/meta/index.html)) which, when invoked, evaluates the concept given the template arguments.

Reply:
Resolved, as it actually referred to line 130. (sorry for the confusion)